appium/lib/screen-object.rb

```diff
@@ -37,127 +36,149 @@ def driver
     wait.until &block
+    sleep 0.2
   end
 
   def wait_step(timeout = 5, message = nil, &block)
     default_wait = driver.default_wait
-    wait = Selenium::WebDriver::Wait.new(:timeout => driver.set_wait(timeout), :message => message)
+    wait = Selenium::WebDriver::Wait.new(timeout: driver.set_wait(timeout), message: message)
     wait.until &block
     driver.set_wait(default_wait)
   end
 
   def enter
-    #pending implementation
-  end
-
-  def scroll_down_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+    driver.send_keys(:enter)
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def gesture(arg)
+    Appium::TouchAction.new(driver).swipe(start_x: arg[0], start_y: arg[1], end_x: arg[2], end_y: arg[3], duration: arg[4]).perform
+  rescue RuntimeError => e
+    raise("Error during gesture \n Error Details: #{e}")
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def scroll(direction = :down, duration = 1000)
+    size = driver.window_size
+    x = size.width / 2
+    y = size.height / 2
+    loc = case direction
+          when :up then    [x, y * 0.5, x, (y + (y * 0.5)), duration]
+          when :down then  [x, y, x, y * 0.5, duration]
+          when :left then  [x * 0.6, y, x * 0.3, y, duration]
+          when :right then [x * 0.3, y, x * 0.6, y, duration]
+          else
+            raise('Only upwards and downwards scrolling are supported')
```

Comment:
Your error message is incorrect. You should change the message to display that `direction` is invalid unless `:up, :down, :left, :right`